controller/cmd/destination/main.go

```diff
@@ -124,3 +141,35 @@ func Main(args []string) {
 	close(done)
 	server.GracefulStop()
 }
+
+func parsePorts(ports string) ([]uint32, error) {
+	var opaquePorts []uint32
+	if ports != "" {
+		for _, portStr := range parsePortRanges(ports) {
+			port, err := strconv.ParseUint(portStr, 10, 32)
+			if err != nil {
+				return nil, err
+			}
+			opaquePorts = append(opaquePorts, uint32(port))
+		}
+	}
+	return opaquePorts, nil
+}
+
+func parsePortRanges(portsString string) []string {
```

Comment:
Why do we return []string here?  Seems like we have ints and then convert them to strings, and then convert them immediately back into ints?

Reply:
Yeah, Unnecessary. Simplified this and also moved this into `util`